src/runtime/internal/Component.ts

```diff
@@ -52,33 +52,33 @@ export function claim_component(block, parent_nodes) {
 	block && block.l(parent_nodes);
 }
 
-export function mount_component(component, target, anchor) {
-	const { fragment, on_mount, on_destroy, after_update } = component.$$;
+export function mount_component({ $$ }, target, anchor) {
+	const { fragment, on_destroy } = $$;
 
 	fragment && fragment.m(target, anchor);
 
 	// onMount happens before the initial afterUpdate
 	add_render_callback(() => {
-		const new_on_destroy = on_mount.map(run).filter(is_function);
+		const new_on_destroy = $$.on_mount.map(run).filter(is_function);
```

Comment:
why destructure `fragment` and `on_destroy`, but not `on_mount` and `after_update`?

Reply:
I destructured `fragment` and `on_destroy` because they are used multiple times in that codeblock.
The reason, only a few properties get destructured is simple: because it needs fewer bytes:

```javascript
// reference
export function mount_component(component, target, anchor) {
	const $$ = component.$$;
	$$.fragment && $$.fragment.m(target, anchor);
   ...
}

// reference - minified
export function m(c, t, a) {
	const $ = c.$;
	$.fragment && $.fragment.m(t, a);
   ...
}

// optimized
export function mount_component({ $$ }, target, anchor) {
	const { fragment } = $$;
	fragment && fragment.m(target, anchor);
   ...
}

// optimized - minified
export function mount_component({ $$: $ }, t, a) {
	const { fragment: f } = $;
	f && f.m(t, a);
   ...
}
```

But it saves only fewer bytes, if the variable gets used multiple times. It will not work for `on_mount`, because it is only accessed a single time:

```javascript
// reference
export function mount_component({ $$ }, target, anchor) {
   ...
	add_render_callback(() => {
		const new_on_destroy = $$.on_mount.map(run).filter(is_function);
      ...
		$$.on_mount = [];
   });
   ...
}

// reference - minified
export function m({ $$: $ }, t, a) {
   ...
	add_render_callback(() => {
		const n = $.on_mount.map(run).filter(is_function);
      ...
		$.on_mount = [];
   });
   ...
}


// with destructuring
export function mount_component({ $$ }, target, anchor) {
   const { on_mount } = $$;
   ...
	add_render_callback(() => {
		const new_on_destroy = on_mount.map(run).filter(is_function);
      ...
		$$.on_mount = [];
   });
   ...
}

// with destructuring - minified
export function m({ $$: $ }, t, a) {
   const { on_mount: o } = $;
   ...
	add_render_callback(() => {
		const n = o.map(run).filter(is_function);
      ...
		$$.on_mount = [];
   });
   ...
}
```
If you count the characters the version with destructuring uses more bytes than the version without destructurung.

I can also destructure every variable to improve code-readability, but this would mean more bytes generated in the output.

Reply:
@tanhauhau should I change the code to destructure all variables for better code readability or leave it like it is for optimal output size?